packages/core/src/shared/ui/nestedWizardPrompter.ts

```diff
@@ -0,0 +1,27 @@
+/*!
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { Wizard, WizardOptions } from '../wizards/wizard'
+import { WizardPrompter } from './wizardPrompter'
+import { createHash } from 'crypto'
+
+export abstract class NestedWizard<T> extends Wizard<T> {
+    // Map to store wizard instances
+    private wizardInstances: Map<string, any> = new Map()
+
+    protected constructor(options: WizardOptions<T>) {
+        super(options)
+    }
+
+    protected createWizardPrompter<T>(constructor: new (...args: any[]) => T, ...args: any[]): WizardPrompter<T> {
```

Comment:
Can we type the createWizardPrompter stronger? Maybe by allowing it to accept only classes that extend Wizard?

Reply:
Added the type enforcement.